### PR TITLE
[Supply] Apk check_superseded_tracks improvements

### PR DESCRIPTION
**Pull request by capyvara**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Improvements to Supply check_superseded_tracks, try to fix and help debugging some users has having, such as in #9574 

### Description
- Added some UI.verbosity messages
- Fixes a bug in which a superseded track may not be computed
- Added tests

### CODE REVIEW

**Comment by andreyslyusar** [comment]
When are you going to merge these changes ?

**Comment by janpio** [comment]
> Fixes a bug in which a superseded track may not be computed
Could you explain that one a bit so a reviewer knows what to look for?

**Comment by capyvara** [comment]
@janpio 
If you merge only the rspec tests the bug will happen (on the remove lesser than the currently being uploaded if it is in an earlier (i.e. alpha) track), the case was that the first found track wasn't considered a candidate for deprecation, fixed by changing a if-else to a if-end.

**Comment by andreyslyusar** [comment]
Can somebody please make a review so we can merge it?

**Comment by KrauseFx** [comment]
Hey everyone, sorry I don't have enough Android knowledge to do a full review of this PR. Maybe someone from @fastlane/android can help out here?

**Comment by haxxpop** [comment]
I really want this one !!

File: supply/lib/supply/uploader.rb [hunk 1]
```diff
@@ -168,34 +168,34 @@ def update_track(apk_version_codes)
...
       config_track_index = tracks.index(Supply.config[:track])
 
       tracks.each_index do |track_index|
-        next if track_index.eql? config_track_index
         track = tracks[track_index]
-
         track_version_codes = client.track_version_codes(track).sort
+        UI.verbose("Found '#{track_version_codes}' on track '#{track}'")
+
+        next if track_index.eql? config_track_index
```
Comment (janpio) [inline supply/lib/supply/uploader.rb]: This was moved down here so the verbose output would still happen?
What exactly does this check? (Maybe add as comment?)

**Comment by janpio** [comment]
@andreyslyusar @haxxpop What do you think of these changes, do they make sense to you?

File: supply/lib/supply/uploader.rb [hunk 2]
```diff
@@ -168,34 +168,34 @@ def update_track(apk_version_codes)
...
-            keep_version_codes = track_version_codes - removed_version_codes
-            max_tracks_version_code = keep_version_codes[0] unless keep_version_codes.empty?
-            client.update_track(track, 1.0, keep_version_codes)
-            UI.message("Superseded track '#{track}', removed '#{removed_version_codes}'")
-          end
+        removed_version_codes = track_version_codes.take_while do |v|
+          v < max_tracks_version_code || (v < max_apk_version_code && track_index > config_track_index)
         end
+
+        next if removed_version_codes.empty?
```
Comment (janpio) [inline supply/lib/supply/uploader.rb]: empty line after this for clarity that the next line is a new thing?

File: supply/lib/supply/uploader.rb (hunk 1 above)
Comment (andreyslyusar) [reply supply/lib/supply/uploader.rb]: We need verbose output in order to get more info why it does failed, more description is in #9574.

File: supply/lib/supply/uploader.rb (hunk 2 above)
Comment (capyvara) [reply supply/lib/supply/uploader.rb]: Not sure, rubocop doesn't complain, feel free to change since it doesn't modify the behavior.

Comment (janpio) [reply supply/lib/supply/uploader.rb]: done.

**Comment by janpio** [comment]
(That message is expected and not a problem, we know that both people (@capyvara and me) have signed the CLA and so can just merge the changed when ready)

**Review by janpio — APPROVED** [review]
Oh, just now I figured out that you @capyvara wrote the initial implementation of this! This makes merging this change a lot easier of course, I expect you know what you were doing back then ;)

**Pull request merged**